### PR TITLE
Validate review

### PR DESCRIPTION
**Pull request by ved0**
* Made some simple review validation on server-side that checks for correct name (no numbers or special characters in the name) and also checks for correct rating (between 0-5). Alerts the client what went wrong, what to change and try again.
*  Made some tests for different types of inputs

OBS! If you are going to test it, dont forget to remove `import fetch from "node-fetch";`

### CODE REVIEW

File: views/partials/addReview.ejs [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
     <div class="add-review-content">
       <form>
-        <label for="name">Name:</label>
+        <label for="name">Namn:</label>
         <input type="text" id="name" name="name" /><br />
 
-        <label for="rating">Rating:</label>
-        <input type="number" min="0" max="5" id="rating" name="rating" /><br />
+        <label for="rating">Betyg:</label>
+        <input type="number" id="rating" name="rating" /><br />
```
Comment (JonathanHolmgren) [inline views/partials/addReview.ejs:19]: Tog du bort min="0" max="5" för att det skulle bli lättare att testa felhantering "checkRating()"?

**Review by JonathanHolmgren — APPROVED** [review]
Snyggt jobbat! Har testat köra detta på server och funkar bra och testarna blev godkända.

File: src/addReviewServer.js [hunk 2]
```diff
@@ -1,15 +1,84 @@
...
+    return true;
+  }
+}
+
+function checkRating(review) {
+  const rating = review.rating;
+  //if rating is null for some reason
+  if (rating == null) {
+    return false;
+  } else if (0 <= rating && rating <= 5 && rating.length > 0) {
```
Comment (Arash-SJPI22) [inline src/addReviewServer.js]: You don't check if the rating has decimals, maybe use Math.round() on the rating before returning it? Because I get a 400 status error in the console, but no error message on the front-end when posting reviews with a decimal rating

File: views/partials/addReview.ejs (hunk 1 above)
Comment (ved0) [reply views/partials/addReview.ejs:19]: I am planning on removing the current input field anyway, so there was no need to keep the min and max, it will already be 5 hats to click, no more, no less :)

File: src/addReviewServer.js (hunk 2 above)
Comment (ved0) [reply src/addReviewServer.js]: Nice catch! Thanks! 👍

File: static/js/addReviewClient.js [hunk 3]
```diff
@@ -1,49 +1,62 @@
-const addReviewButton = document.querySelector(".add-review-button");
-let addReviewContent = document.querySelector(".add-review-content");
+let addReviewButton = document.querySelector(".add-review-button");
+const addReviewContent = document.querySelector(".add-review-content");
```
Comment (saeraa) [inline static/js/addReviewClient.js]: Why did you change to let? 🤔

File: tests/addReviewServer.test.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  createdAt: new Date(),
+  updatedAt: new Date(),
+  createdBy: "Vedad",
+  updatedBy: "Vedad",
+};
+
+describe("Tests of the validateReview function on both correct and incorrect input)", () => {
+  test("Wrong input (name)", () => {
+    const result = validateReview(mockWrongNameReview);
+    expect(result.validated == false).toBeTruthy();
```
Comment (richardolsson) [inline tests/addReviewServer.test.js:79]: The way this expectation is expressed confuses me. I would do `expect(result.validated).toBeFalsy()` instead.

File: src/addReviewServer.js [hunk 5]
```diff
@@ -1,15 +1,82 @@
...
+      return {
+        validated: true,
+        message: "Tack för din recension, " + review.author + "!",
+        status: 201,
+      };
+    } else {
+      return {
+        validated: false,
+        message: "Betyget behöver vara mellan 0-5!",
+        status: 422,
```
Comment (richardolsson) [inline src/addReviewServer.js:39]: I would expect the normal 400 status code in all of these cases. I'm not saying 422 is wrong, but it's unexpected. 😊

File: src/addReviewServer.js [hunk 6]
```diff
@@ -1,15 +1,82 @@
...
+      message: "Skriv in ett korrekt namn!",
+      status: 422,
+    };
+  }
+}
+
+function checkAuthor(review) {
+  const author = review.author;
+  //I kept this because it feels like a cool idea to spin upon.
+  const regex = /[^a-zåäöA-ZÅÄÖ -]/;
```
Comment (richardolsson) [inline src/addReviewServer.js:60]: What happens if your name is Bjørn, Günter, Matthíldur or something in Korean? I always recommend against this kind of validation.

**Review by richardolsson — COMMENTED** [review]
I've left some comments below, but all in all this is very good work!
